scripts/parse_nagim.py

```diff
@@ -0,0 +1,467 @@
+    # corresponding TBI/CRAM, and initializing the sample list.
+    _find_upload_files(
+        upload_gvcf_bucket,
+        upload_cram_bucket,
+        tmp_dir,
+        overwrite=True,  # Force finding files
+    )
+
+
+@cli.command()
+@click.argument('sample_to_project_path')
+@click.option('--prod', 'prod', is_flag=True)
+@click.option('--tmp-dir', 'tmp_dir')
+@click.option(
+    '--confirm', is_flag=True, help='Confirm with user input before updating server'
+)
+@click.option('--dry-run', 'dry_run', is_flag=True)
+@click.option(
+    '--skip-checking-objects',
+    'skip_checking_objects',
+    is_flag=True,
+    help='Do not check objects on buckets (existence, size, md5)',
+)
+def parse(
+    sample_to_project_path: str,
+    prod,
+    tmp_dir,
+    confirm: bool,
+    dry_run: bool,
+    skip_checking_objects: bool,
+):
+    """
+    Assuming the data is transferred to the CPG bucket, populat the SM project.
+    """
+    namespace = 'main' if prod else 'test'
+
+    upload_gvcf_bucket = f'gs://cpg-{PROJECT}-{namespace}-upload/gvcf'
+    upload_cram_bucket = f'gs://cpg-{PROJECT}-{namespace}-upload/cram'
+
+    # Finds GVCFs and CRAMs after transferring, and checks that all of them have
+    # corresponding TBI/CRAM, and initializing the sample list.
+    samples = _find_upload_files(upload_gvcf_bucket, upload_cram_bucket, tmp_dir)
+
+    # Checks sample and project IDs.
+    # Some samples processed with Terra use CPG IDs, checking if we already
+    # have them in the SMDB and fixing the external IDs.
+    samples = _fix_ids(samples, sample_to_project_path, prod=prod)
+
+    # Creating a parser for each project separately
+    for proj_id in PROJECT_ID_MAP.values():
+        sm_proj_id = proj_id if prod else f'{proj_id}-test'
+        sample_tsv_file = join(tmp_dir, f'samples-{sm_proj_id}.csv')
+        df = pd.DataFrame(
+            dict(
+                cpg_id=s.cpg_id,
+                ext_id=s.ext_id,
+                gvcf=s.gvcf,
+                cram=s.cram,
+            )
+            for s in samples
+            if s.project_id == proj_id
+        )
+        if len(df) == 0:
+            logger.info(f'No samples for project {sm_proj_id} found, skipping')
+            continue
+
+        df.to_csv(sample_tsv_file, index=False)
+        logger.info(
+            f'Processing {len(df)} samples for project {sm_proj_id}, '
+            f'sample manifest: {sample_tsv_file}'
+        )
+
+        parser = NagimParser(
+            path_prefix=None,
+            sample_metadata_project=sm_proj_id,
+            skip_checking_gcs_objects=skip_checking_objects,
+            confirm=confirm,
+            verbose=False,
+        )
+        with open(sample_tsv_file) as f:
+            parser.parse_manifest(f, dry_run=dry_run)
+
+
+def _fix_ids(
+    samples_from_found_files: List[Sample],
+    sample_to_project_path: str,
+    prod: bool,
+) -> List[Sample]:
+    sapi = SampleApi()
```

Comment:
Docstring please

Reply:
Didn't want to duplicate the comment where the function is called, but will add

Reply:
Actually, split it into 2 methods, one to fix sample IDs, another to set project IDs.